app/forms/providers/link_application/find_link_application_form.rb

```diff
@@ -34,7 +34,8 @@ def exclude_from_model
       end
 
       def save
-        attributes[:lead_application_id] = @found_application&.id
+        attributes[:target_application_id] = @found_application&.id
+        attributes[:lead_application_id] = @found_application&.lead_application&.id.presence || @found_application&.id
```

Comment:
Discussion: I like the safe nav operator, however, the chaining of them here feels off. As we don't have a guide/standard, let's discuss.

Reply:
Rubocop enforces it.  If you use a safe navigation operator on the first object you _have_ to use it on all subsequent calls.  If you remove the second one, you get this 
```shell
Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.
        attributes[:lead_application_id] = @found_application&.lead_application.id.presence || @found_application&.id
                                                                               ^^^
```